std/bigint.d

```diff
@@ -1761,3 +1761,90 @@ unittest
     auto b = immutable BigInt("123");
     assert(b == 123);
 }
+
+// Issue 18086
+void divMod(const BigInt x, const BigInt y, out BigInt quotient, out BigInt remainder)
+{
+    BigUint q, r;
+    BigUint.divMod(x.data, y.data, q, r);
+    quotient.sign = x.sign != y.sign;
+    quotient.data = q;
+    remainder.sign = x.sign;
+    remainder.data = r;
+}
+
+void divMod(T1, T2)(T1 x, T2 y, out BigInt quotient, out BigInt remainder)
+    if (is(T1 : const BigInt) || is(T2 : const BigInt) && is(typeof(BigInt(x))) && is(typeof(BigInt(y))))
+{
+    return divMod(BigInt(x), BigInt(y), quotient, remainder);
```

Comment:
static if this to only create a new big int when it’s not already one

Reply:
Since BigInt is COW, the constructor call is basically free (it's an array assignment and a bool assignment - no copying takes place). Using static if to conditionally call the constructor complicates the code to no benefit.

Reply:
However, creating two overloads on the form `divMod(T)(T x, const BigInt y, ...) if (is(typeof(BigInt(x))))` is more readable than the current template constraint.